mvc/models/model-binding.rst

```diff
@@ -1,8 +1,105 @@
+
+.. note:: When the binder finds the first matching name in the data source, it stops looking for more elements matching that name, and moves onto the next property or item in the parameter list. 
+ 
+Additionally, there are some special data types that ASP.NET MVC must consider when performing model binding:
+
+- ``Request.Files``: Uploaded files that are part of the HTTP request.
+- ``IFormCollection``: The collection of key-value pairs that represent the form data.
+- Cancelation token: Used to cancel activity in asynchronous controllers.
+
+Once model binding is complete, validation occurs. The default model binders are quite smart, and work great for the vast majority of development scenarios. However, ASP.NET MVC is extensible so if you have unique needs you can customize the built in model binders as you wish.  
+
+Customize default model binding behavior with attributes 
+--------------------------------------------------------
+ASP.NET MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source, which is what the ``[From*]`` attributes do, such as the ``[FromForm]`` or ``[FromQuery]``. While the topic of model binding extensibility in depth is outside of the scope of this doc, here are a few customizations that you should be aware of:
+
+- ``[BindRequired]``: This attribute forces a Model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: Only supported on action parameters. 
+- ``[FromBody]``: Instructs the model binder to use the JSON media type formatter rather than regular model binding.
+- ``[ModelBinder]``: You reject the current model binder and substitute it with your own.
+
+A good example of binding attribute usage in the wild is when you see the ``[FromBody]`` attribute in action methods of API Controllers. While there are only a few differences between standard MVC and API Controllers, one such difference is that API Controllers attempt to retrieve data from the request URI in PUT and POST messages, rather than from the body by default. This means that when a page sends data via an Ajax call, Web API actions still look in the header, but applying the ``[FromBody]`` attribute will direct the model binder to the correct data source, as shown in the snippet below:
```

Comment:
I think you can remove this paragraph. It doesn't really make sense. Instead you could give an example of using FromBody to bind JSON data to a complex type.